q.js

```diff
@@ -1012,6 +1027,14 @@ function trackRejection(promise, reason) {
     if (!trackUnhandledRejections) {
         return;
     }
+    if(typeof process === "object" && typeof process.emit === "function"){
+        Q.nextTick.runAfter(function(){
+            if(array_indexOf(unhandledRejections, promise) !== -1){
+                process.emit("unhandledRejection", reason, promise);
```

Comment:
The `process` bits are part of the specification for node/io and work this way across promise libraries and soon native promises in io.js

Firing events for browsers here is something we can definitely wait with - I asked @domenic if he wanted it and he said he did. I can leave it just like that if you'd like.